parser.cpp

```diff
@@ -1185,9 +1194,9 @@ namespace Sass {
     { return new (ctx.mem) Null(pstate); }
 
     if (lex< identifier >()) {
-      String_Constant* str = new (ctx.mem) String_Constant(pstate, lexed);
+      String_Constant* str = new (ctx.mem) String_Quoted(pstate, lexed);
```

Comment:
> it means the parsed string might has been a quoted string

The semantics here are confusing to me. `String_Constant` is an unquoted string where as `String_Quoted` is either a quoted or unquoted string?

Feels like there should be a higher abstraction that given a string turns it into either a `String_Constant` or an `String_Quoted`?
Or maybe `String_Quoted` is a bad name if the string isn't necessarily quoted?

My concern is not knowing which type to use in any one circumstance.